include/seqan3/alignment/pairwise/edit_distance_unbanded.hpp

```diff
@@ -407,7 +400,9 @@ class edit_distance_unbanded_semi_global_policy :
     size_t back_coordinate_first() const noexcept
     {
         derived_t const * self = static_cast<derived_t const *>(this);
-        return std::ranges::distance(std::ranges::begin(self->database), _best_score_col) + 1u;
+        // offset == 0u is a special case if database sequence is empty, because in this case the best column is zero.
+        size_t offset = std::ranges::begin(self->database) == self->database_it_end ? 0u : 1u;
```

Comment:
Isn't the condition equivalent to `std::ranges::empty(self->database)` ?

Reply:
true, thank you.